demos/common/python/models/segmentation.py

```diff
@@ -17,42 +17,25 @@
 import cv2
 import numpy as np
 
-from .model import Model
+from .image_model import ImageModel
 from .utils import load_labels
 
 
-class SegmentationModel(Model):
-    def __init__(self, ie, model_path, labels=None):
-        super().__init__(ie, model_path)
-
-        self.input_blob_name = self.prepare_inputs()
-        self.out_blob_name = self.prepare_outputs()
+class SegmentationModel(ImageModel):
+    def __init__(self, ie, model_path, input_transform=None, resize_type='default',
+                 labels=None):
+        super().__init__(ie, model_path, input_transform=input_transform, resize_type=resize_type)
 
         if isinstance(labels, (list, tuple)):
             self.labels = labels
         else:
             self.labels = load_labels(labels) if labels else None
 
-    def prepare_inputs(self):
-        if len(self.net.input_info) != 1:
-            raise RuntimeError("Demo supports topologies only with 1 input")
-
-        blob_name = next(iter(self.net.input_info))
-        blob = self.net.input_info[blob_name]
-        blob.precision = "U8"
-        blob.layout = "NCHW"
-
-        input_size = blob.input_data.shape
-        if len(input_size) == 4 and input_size[1] == 3:
-            self.n, self.c, self.h, self.w = input_size
-        else:
-            raise RuntimeError("3-channel 4-dimensional model's input is expected")
+        self.output_blob_name = self._get_outputs()
```

Comment:
```suggestion
        self.out_blob_name = self._get_outputs()
```
AttributeError: 'SegmentationModel' object has no attribute 'out_blob_name'

Reply:
Done. 
Fixes below